### ISSUE TITLE
std/esm with babel and ava

### ISSUE THREAD

**Issue by dacz** [standard-things/esm#193]
ref: https://github.com/avajs/ava/issues/1590

I prepared small repo

https://github.com/dacz/esm-ava-example

where are branches working smoothly (generally flow, ava, no babel).

Branch `with-flow-with-babel7` and `with-flow-with-babel6` demonstrates issue. Using babel, it seems that `ava` setting in package `"require": ["@std/esm", "@babel/register"],` are not enough to transform imports:

```
/Users/dacz/development/esmava/giveMeNanoid.test.js:4
import giveMeNanoid from './giveMeNanoid';
^^^^^^

SyntaxError: Unexpected token import
```

Once adding `preset-env` to babelrc fixes the issue (at least tested in babel6) but this is not what I want to do.

Seems I'm unable to make it work with babel (6 or 7). Probably some faux-pas mistake on my side... 

I'd be happy to help with more experiments.

(and I'm really thankful for tools like std/esm and ava)

**Comment by dnalborczyk** [comment]
@dacz   I haven't looked into your issue, but I'm quite certain Babel 6.x is not supported by std/esm. Only Babel 7+ is supported (not sure which minor beta version, but definitely the latest)

**Comment by dacz** [comment]
@dnalborczyk thanks for info, I didn't know this. But still even with the babel7 the error about `import` is still there.

**Comment by dacz** [comment]
as @dnalborczyk wrote, I focused on the `babel7` branch.

`node -r @std/esm -r @babel/register useNanoid.js` works like a charm.
`npm run test` fails with

```
/Users/dacz/development/esmava/giveMeNanoid.test.js:4
import giveMeNanoid from './giveMeNanoid';
^^^^^^

SyntaxError: Unexpected token import
```

I tried to patch ava (as mentioned in https://github.com/avajs/ava/issues/1575 by @vjpr), still the same error. (btw: ava patched with this hack sorta works in other project for me with babel7).

And `babel giveMeNanoid.js` transpiles the file correctly - leaves import as is and strips the flow annotation.

**Comment by dnalborczyk** [comment]
@dacz  I got your sample repo (branch: with-flow-with-babel7) running with some small changes:

package.json
```
  "ava": {
    "files": [
      "**/**/*.test.js"
    ],
    "require": [
      "@babel/register",
      "@std/esm"
    ]
  },
```

notice: I switched the require order, and removed babelrc: true.

_The following is just an assumption, not sure if it is true:_
I think that babelrc: _true_ or _inherit_ would not work because of version mismatch. It appears that Ava is using Babel 6 under the hood to make things "just work", whereas in your project code you want to use Babel 7 with std/esm -- so using the Babelrc is not going to work.

Not quite sure right now what you would need to do in order to use e.g. flow-types in your test files.
